client/src/app/appCtrl.js

```diff
@@ -202,81 +311,51 @@ angular.module('app').controller('appCtrl', ['$scope', '$http', '$timeout', 'app
         $scope.view.loading = false;
         $scope.$apply();
       },
-      error: function(error) {
-        $scope.view.loading = false;
-        handleError("Could not download file.", "danger");
-      }
-    });
-  };
 
-  var streamRemoteFile = function(url) {
-    resetFields();
-    Papa.RemoteChunkSize = appConfig.REMOTE_CHUNK_SIZE;
-    var firstChunkComplete = false;
-    Papa.parse(url, {
-      download: true,
-      skipEmptyLines: true,
-      header: true,
-      dynamicTyping: true,
-      worker: false, // multithreaded, !but does NOT work with other libs in app.js or streaming
-      comments: "#",
+      // used for 'stream*' mode
       chunk: function(chunk, parser) {
+        iter++;
         if (!firstChunkComplete) {
           streamParser = parser;
           loadedFields = generateFieldMap(chunk.data, appConfig.EXCLUDE_FIELDS);
           firstChunkComplete = true;
         }
-        loadData(chunk.data);
-      },
-      beforeFirstChunk: function(chunk) {
-        $scope.view.loadedFileName = getRemoteFileName(url);
-        var rows = chunk.split(/\r\n|\r|\n/);
-        rows.splice(1, appConfig.HEADER_SKIPPED_ROWS);
-        $scope.view.loading = false;
-        return rows.join('\n');
-      },
-      //fastMode: true, // automatically enabled if no " appear
-      error: function(error) {
-        handleError("Could not stream file.", "danger");
-        $scope.view.loading = false;
-      }
-    });
-  };
 
-  // read and parse a CSV file
-  var streamLocalFile = function(file) {
-    resetFields();
-    Papa.LocalChunkSize = appConfig.LOCAL_CHUNK_SIZE; // set this to a reasonable size
-    var firstChunkComplete = false;
-    Papa.parse(file, {
-      skipEmptyLines: true,
-      header: true,
-      dynamicTyping: true,
-      worker: false, // multithreaded, !but does NOT work with other libs in app.js or streaming
-      comments: "#",
-      chunk: function(chunk, parser) {
-        if (!firstChunkComplete) {
-          streamParser = parser;
-          loadedFields = generateFieldMap(chunk.data, appConfig.EXCLUDE_FIELDS);
-          firstChunkComplete = true;
+        // as files grow very large in continuous monitoring
+        // for speed reasons (on large files) in the online monitor mode, 
+        // we skip renderning all but the last window
+        // this technique only helps after the file was once read to the end (completed)
+        if(iter <= $scope.view.monitor.lastChunkIter && iter > 1) {
```

Comment:
here and below are the optimization to avoid parsing the whole large file on updates in monitoring mode.